### PR TITLE
chore(component): minor table updates

### PR DESCRIPTION
**Pull request by bc-apostoliuk**
## What?
1. Update table header row paddings;
2. Update table row paddings;
3. Update table header background-color
4. Update table header font color

## Why?

To follow the new guideline;

## Screenshots/Screen Recordings
Screenshot 2022-04-27 at 15 11 56

## Testing/Proof

Locally.

### CODE REVIEW

**Review by golcinho — APPROVED** [review]
LGTML 👍 Please wait for approval from @bigcommerce/product-design

**Review by vdm-kr — APPROVED** [review]
All good 👌

**Pull request merged**
